android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/SelectLanguageActivity.java

```diff
@@ -0,0 +1,111 @@
+  private static final String titleKey = "title";
+  private static final String subtitleKey = "subtitle";
+  private static final String iconKey = "icon";
+  private static Context context;
+
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    supportRequestWindowFeature(Window.FEATURE_NO_TITLE);
+    context = this;
+
+    setContentView(R.layout.activity_list_layout);
+    final Toolbar toolbar = findViewById(R.id.list_toolbar);
+    setSupportActionBar(toolbar);
+    getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+    getSupportActionBar().setDisplayShowHomeEnabled(true);
+    getSupportActionBar().setDisplayShowTitleEnabled(false);
+
+    final ListView listView = findViewById(R.id.listView);
+    listView.setFastScrollEnabled(true);
+
+    final Keyboard keyboard = (Keyboard)getIntent().getSerializableExtra("keyboard");
+    final String packageID = keyboard.getPackageID();
+    final String keyboardID = keyboard.getKeyboardID();
+    final String keyboardName = keyboard.getKeyboardName();
+    String title = String.format(getString(R.string.title_select_language_for_package), keyboardName);
+    final TextView textView = findViewById(R.id.bar_title);
+    textView.setText(title);
+    if (titleFont != null) {
+      textView.setTypeface(titleFont, Typeface.BOLD);
+    }
+
+    // Get the list of available languages from kmp.json
+    File resourceRoot =  new File(context.getDir("data", Context.MODE_PRIVATE).toString() + File.separator);
+    PackageProcessor kmpProcessor =  new PackageProcessor(resourceRoot);
+    List<Keyboard> availableKeyboardsList = kmpProcessor.getLanguageList(packageID, keyboardID);
+
+    list = new ArrayList<HashMap<String, String>>();
+    for (Keyboard k : availableKeyboardsList) {
+      final String noIcon = "0";
+      HashMap<String, String> hashMap = new HashMap<>();
+      hashMap.put(titleKey, k.getLanguageName());
+      hashMap.put(subtitleKey, k.getLanguageID());
+      String icon = String.valueOf(R.drawable.ic_arrow_forward);
+      hashMap.put(iconKey, icon);
+      list.add(hashMap);
+    }
+
+    String[] from = new String[]{titleKey, subtitleKey, iconKey};
+    int[] to = new int[]{com.tavultesoft.kmea.R.id.text1, com.tavultesoft.kmea.R.id.text2, com.tavultesoft.kmea.R.id.image1};
+
+    ListAdapter adapter = new SimpleAdapter(context, list, com.tavultesoft.kmea.R.layout.list_row_layout2, from, to) {
+    };
+    listView.setAdapter(adapter);
+    listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+
+      @Override
+      public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+        HashMap<String, String> hashMap = (HashMap<String, String>) parent.getItemAtPosition(position);
+        Keyboard k = availableKeyboardsList.get(position);
+        KMManager.addKeyboard(context, k);
+        Toast.makeText(context, "Keyboard added", Toast.LENGTH_LONG).show();
```

Comment:
This should be localizable and it should also show `"Added language <y> for keyboard <x>"`? or something like that?